### ISSUE TITLE
*.db.gz made from mksql3db.sh no longer recognizable by GTFSOffline app

### ISSUE THREAD

**Issue by joelmaxuel** [wbrenna/GTFSOffline#25]
Symptom: GTFSOfline will scan Download folder, but database does not appear in app.

Environment:
Motorola X Play (Lux) running custom 7.1.2 firmware (problem noticed after a /sdcard wipe).
Could be possible Debian 9 gzip added a feature to format to break readability, or (admittedly) gzip on phone unavailable/broken.

gzip on PC to compile *.db.gz is 1.6

Partial logcat to come to this conclusion:

    01-24 15:50:19.251 W/System  (9639): ClassLoader referenced unknown path: /data/app/com.wbrenna.gtfsoffline-1/lib/arm
    ...
    01-24 15:50:26.881 V/UpdateActivity(9639): Download folder is: Download
    01-24 15:50:26.885 V/UpdateActivity(9639): File found: halifax.db.gz
    01-24 15:50:26.886 V/UpdateActivity(9639): halifax.db.gz
    01-24 15:50:26.888 E/UpdateActivity(9639): IOException exception
    01-24 15:50:26.888 W/System.err(9639): java.util.zip.ZipException: Not in GZIP format
    01-24 15:50:26.888 W/System.err(9639): 	at java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:163)
    01-24 15:50:26.888 W/System.err(9639): 	at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:77)
    01-24 15:50:26.889 W/System.err(9639): 	at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:89)
    01-24 15:50:26.892 W/System.err(9639): 	at com.wbrenna.gtfsoffline.UpdateActivity$DBMover.doInBackground(UpdateActivity.java:153)
    01-24 15:50:26.892 W/System.err(9639): 	at com.wbrenna.gtfsoffline.UpdateActivity$DBMover.doInBackground(UpdateActivity.java:95)
    01-24 15:50:26.892 W/System.err(9639): 	at android.os.AsyncTask$2.call(AsyncTask.java:305)
    01-24 15:50:26.892 W/System.err(9639): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
    01-24 15:50:26.892 W/System.err(9639): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
    01-24 15:50:26.892 W/System.err(9639): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
    01-24 15:50:26.892 W/System.err(9639): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
    01-24 15:50:26.892 W/System.err(9639): 	at java.lang.Thread.run(Thread.java:761)
    01-24 15:50:26.910 V/UpdateActivity(9639): Finished copying stops!
    01-24 15:50:26.938 I/ActivityManager(768): Displayed com.wbrenna.gtfsoffline/.UpdateActivity: +100ms

**Comment by joelmaxuel** [comment]
The issue, depending on how the db is staged from Download, could be that a corrupted copy is stuck and unable to be replaced.  I renamed halifax.db.gz to a different name (same extensions), and the new name is picked up in logcat but so are the same errors.

More importantly, I backed up my system and flashed the last known good ROM version (the one that I had successfully set up GTFSOffline with before).  Same errors.

I had considered a bug in zip util (what I assume to be in framework_base) since the custom ROM had a rapid run update of security bulletins to the end of 2018 and fwb was part of the update (however did not see any commits related to zip).

Based on the tests thus far, it does not seem the security patches caused an issue but I can test this by formatting the sdcard, and try a clean install with older and newer ROM revisions.  But I would rather check for cruft first, stuff that can last between clearing GTFSOffline app cache/data and reinstalling.  But I don't know where all the files are located.  The apps data folder was empty last I looked (even after changing settings).  Suggestions?
